docs/topics/catalyst/catalyst.md

```diff
@@ -0,0 +1,142 @@
+
+## Model
+
+G&C are ERC-20. Each catalyst type is a different smart contract (ie Catalyst RARE is a smart contract, Catalyst Legendary another...). Like catalyst, each gem type is a different ERC-20 smart contract (ie GemsPower...). Internally every contract is identified by the Ids defined above.
+
+The **GemsCatalystsRegistry** contract manage every G&C contract, it can add new G&C contract, handle Id's, burn tokens if needed and expose attributes of an asset...
+
+
+### Class diagram
+```plantuml
+title class diagram
+interface IGemsCatalystsRegistry {
+    
+}
+
+class GemsCatalystsRegistry #palegreen{
++ getAttributes(...)
++ getMaxGems(...)
++ burnDifferentGems(...)
++ burnDifferentCatalysts(...)
++ batchBurnGems(...)
++ batchBurnCatalysts(...)
++ addGemsAndCatalysts(...)
++ doesGemExist(...)
++ burnCatalyst(...)
++ burnGem(...)
+}
+note top of ERC2771Context : "Meta-transaction handler"
+abstract ERC2771Context {}
+class WithSuperOperators {} 
+class WithAdmin {}
+class Gem #palegreen{
+    + uint16 gemId
+}
+class DefaultAttributes{}
+class Catalyst #palegreen{
+    + uint16 catalystId
+    - uint8  _maxGems
+    +  getAttributes(...)
+}
+interface IAttributes
+class ERC20Token
+
+IGemsCatalystsRegistry <|.. GemsCatalystsRegistry
+ERC2771Context <|-- GemsCatalystsRegistry
+WithSuperOperators <|-- GemsCatalystsRegistry
+WithAdmin<|--WithSuperOperators
+ERC20Token<|--Gem
+ERC20Token<|--Catalyst
+IAttributes<|..Catalyst
+DefaultAttributes -- Catalyst
+IAttributes <|.. DefaultAttributes
+Gem o-- GemsCatalystsRegistry
+Catalyst o--GemsCatalystsRegistry
+```
+
+## Processes
+### Get every attribute of gems on a catalyst in an asset
+
+An asset could fetch all the gems data of a catalyst it's currently owning. It's using GemsCatalystRegistry.GetAttributes. All the Asset part is out of scope, see Asset documentation.
+
+```plantuml
+
+title sequence diagram
+
+
+entity GemsCatalystsRegistry
+entity Catalyst
+entity DefaultAttributes
+
+
+note right of GemsCatalystsRegistry
+An asset has to be minted with G&C. 
+A CatalystApplied events are then created. 
+The events param is retrieve from these events.
+end note
+
+-> GemsCatalystsRegistry: getAttributes(catalystId, assetId, events[]) 
+GemsCatalystsRegistry -> Catalyst: getAttributes(assetId, events[])
+Catalyst -> DefaultAttributes: getAttributes(assetId, events[])
+note over  DefaultAttributes: Random values are calculated using gems data in event[]  
+Catalyst <-- DefaultAttributes: values []
+GemsCatalystsRegistry <-- Catalyst: values []
+<--GemsCatalystsRegistry: values []
+
+```
+### Burn 1 LEGENDARY and 1 EPIC Catalyst in batch
+For this example we assume that an asset want to burn two catalysts in a single tx. All the Asset part is out of scope, see Asset documentation. 
+
+```plantuml
+
+entity GemsCatalystsRegistry
+entity "LEGENDARY Catalyst" as L
+entity "EPIC Catalyst" as E
+->GemsCatalystsRegistry: BatchBurnCatalysts(from,  catalystIds[4,3], amounts[1,1])
```

Comment:
Add an user
`user->GemsCatalystsRegistry`

Reply:
Actually it's the asset that call this function, I don't think the user could call it directly.

Reply:
Ok then you can call it Asset contract for example

Reply:
The problem is plantUML add a new "entity" call asset contract by default. I also do the same in the burn sequence diagram an arrow from "nothing" line 149.